src/pre_defined_classes/mathematical_program/hard/q4/CalcBillTest.java

```diff
@@ -0,0 +1,26 @@
+package pre_defined_classes.mathematical_program.hard.q4;
+
+import global.BaseTest;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CalcBillTest extends BaseTest {
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("With tips, the amount you need to pay is: "),
+                new DoubleLiteral(28.0,28.0, "math")
+        };
+    }
+
+    public void runMain() {
+        CalcBill.main(new String[]{});
+    }
+
+    @Test
+    public void tipPercentageTest() {
+        assertEquals("28.0", getItemByName("math"), "Your math is incorrect");
```

Comment:
This test would never fail, as Clause only accepts 28, so it is redundant if the potential values are not widened

Reply:
It's more about providing better feedback than explicitly needing to check the math. If the math is wrong but I didn't have the test they might think their format is the only thing that's wrong.